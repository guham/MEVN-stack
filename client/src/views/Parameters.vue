<template>
  <div class="text-center">
    <h1 class="pb-2">{{ $t('parameters') }}</h1>
    <div class="flex justify-center pt-6">
      <form class="w-full max-w-xs">
        <div class="md:flex md:items-center mb-6">
          <div class="md:w-1/3">
            <label
              class="block font-bold md:text-right mb-1 md:mb-0 pr-4"
              for="locale-changer">{{ $t('language') }}</label>
          </div>
          <div class="md:w-2/3">
            <locale-changer/>
          </div>
        </div>
      </form>
    </div>
  </div>
</template>

<i18n>
{
  "en": {
    "parameters": "Parameters",
    "language": "Language"
  },
  "fr": {
    "parameters": "Paramètres",
    "language": "Langue"
  },
  "es": {
    "parameters": "Parámetros",
    "language": "Idioma"
  }
}
</i18n>

<script>
import LocaleChanger from '@/components/LocaleChanger.vue';

export default {
  name: 'Parameters',
  components: {
    LocaleChanger,
  },
};
</script>
