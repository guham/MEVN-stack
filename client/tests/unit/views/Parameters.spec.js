import Parameters from '@/views/Parameters.vue';
import factory from '../factory';

describe('Parameters.vue', () => {
  test('use LocaleChanger component', () => {
    const wrapper = factory(Parameters);
    const { LocaleChanger } = wrapper.vm.$options.components;
    expect(LocaleChanger).toBeDefined();
  });

  test('has the expected html structure', () => {
    const wrapper = factory(Parameters);
    const template = wrapper.html();
    expect(template).toMatchSnapshot();
  });
});
