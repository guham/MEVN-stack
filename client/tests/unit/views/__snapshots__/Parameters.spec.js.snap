// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parameters.vue has the expected html structure 1`] = `
<div>
  <h1 class="pb-2">parameters</h1>
  <div class="flex justify-center pt-6">
    <form class="w-full max-w-xs">
      <div class="md:flex md:items-center mb-6">
        <div class="md:w-1/3"><label for="locale-changer" class="block font-bold md:text-right mb-1 md:mb-0 pr-4">language</label></div>
        <div class="md:w-2/3">
          <localechanger-stub></localechanger-stub>
        </div>
      </div>
    </form>
  </div>
</div>
`;
